src/main/java/frc/robot/swerve/ModuleIOTalonFX.java

```diff
@@ -10,53 +11,80 @@
 import edu.wpi.first.math.kinematics.SwerveModulePosition;
 import edu.wpi.first.math.kinematics.SwerveModuleState;
 import edu.wpi.first.wpilibj.DutyCycleEncoder;
-import edu.wpi.first.wpilibj2.command.Command;
-import edu.wpi.first.wpilibj2.command.RunCommand;
+import lib.PhoenixOdometryThread;
 import lib.math.AngleUtil;
 import lib.math.differential.Integral;
 import lib.units.Units;
 
-public class ModuleIOReal implements ModuleIO {
+import java.util.List;
+import java.util.Queue;
+
+public class ModuleIOTalonFX implements ModuleIO {
 
     private final TalonFX driveMotor;
     private final TalonFX angleMotor;
     private final DutyCycleEncoder encoder;
 
+    private final TalonFXConfiguration driveConfig;
+    private final TalonFXConfiguration angleConfig;
+
     private final Integral driveSupplyChargeUsedCoulomb = new Integral();
     private final Integral driveStatorChargeUsedCoulomb = new Integral();
     private final Integral angleSupplyChargeUsedCoulomb = new Integral();
     private final Integral angleStatorChargeUsedCoulomb = new Integral();
-    private final MotionMagicVoltage angleControlRequest =
-            new MotionMagicVoltage(0).withEnableFOC(true).withSlot(0);
+    private final PositionVoltage angleControlRequest =
```

Comment:
Should use the new Motion Magic Expo mode
https://pro.docs.ctr-electronics.com/en/stable/docs/api-reference/device-specific/talonfx/motion-magic.html#motion-magic-expo

Reply:
We tried using the motion magic expo, and we tried normal motion magic. But in the end we found that regular position control is best.